src/main/scala/org/scalameter/Bench.scala

```diff
@@ -22,49 +21,91 @@ abstract class Bench[U] extends DSL[U] with Serializable {
 
 object Bench {
 
-  @deprecated("Please use Bench.Quick instead", "0.7")
-  type Quickbenchmark = Quick
-  
-  /** Quick benchmark run in the same JVM.
+  /** Benchmark that runs snippets in the same JVM.
    *  Reports result into the console.
    */
-  abstract class Quick extends Bench[Double] {
-    def executor: Executor[Double] = new execution.LocalExecutor(
-      Warmer.Default(),
-      Aggregator.min,
+  abstract class Local[U: Pickler] extends Bench[U] {
+    def warmer: Warmer = new Warmer.Default
+
+    def aggregator: Aggregator[U]
+
+    def executor: Executor[U] = new execution.LocalExecutor(
+      warmer,
+      aggregator,
       measurer
     )
-    def measurer: Measurer[Double] = new Measurer.Default
-    def reporter: Reporter[Double] = new reporting.LoggingReporter
+
     def persistor: Persistor = Persistor.None
+
+    def reporter: Reporter[U] = new reporting.LoggingReporter
+  }
+
+  /** Benchmark that runs snippets in separate JVMs.
+   *  Reports result into the console.
+   */
+  abstract class Forked[U: Pickler: PrettyPrinter] extends Bench[U] {
+    def warmer: Warmer = new Warmer.Default
+
+    def aggregator: Aggregator[U]
+
+    def executor: Executor[U] = new execution.SeparateJvmsExecutor(
+      warmer,
+      aggregator,
+      measurer
+    )
+
+    def persistor: Persistor = Persistor.None
+
+    def reporter: Reporter[U] = new reporting.LoggingReporter
+  }
+
+  /** Benchmark that runs snippets in separate JVMs.
+   *  It persists results in the gzipped JSON format with appropriate reporter.
+   */
+  abstract class Persisted[U: Pickler: PrettyPrinter] extends Bench[U] {
+    def warmer: Warmer = new Warmer.Default
+
+    def aggregator: Aggregator[U]
+
+    def executor: Executor[U] = new execution.SeparateJvmsExecutor[U](
+      warmer,
+      aggregator,
+      measurer
+    )
+
+    def persistor: Persistor = new persistence.GZIPJSONSerializationPersistor
+  }
+
+  @deprecated("Please use Bench.Quick instead", "0.7")
+  type Quickbenchmark = Quick
+  
+  /** Quick benchmark runs snippets in the same JVM.
+   *  Reports execution time into the console.
+   */
+  abstract class Quick extends Local[Double] {
```

Comment:
I'm thinking we should call this `LocalTime` and the other one `ForkedTime` to be more intuitive and consistent.